google-cloud-spanner/src/main/java/com/google/cloud/spanner/StructReader.java

```diff
@@ -361,6 +373,38 @@ default Value getValue(String columnName) {
    */
   List<Long> getLongList(String columnName);
 
+  /**
+   * @param columnIndex index of the column
+   * @return the value of a non-{@code NULL} column with type {@code Type.array(Type.float32())}.
+   * @throws NullPointerException if any element of the array value is {@code NULL}. If the array
+   *     may contain {@code NULL} values, use {@link #getFloatList(int)} instead.
+   */
+  float[] getFloatArray(int columnIndex);
```

Comment:
Same as above: These should have a default implementation that just throws an error.

Reply:
Done